### PR TITLE
Enable android-macos build

### PR DESCRIPTION
**Pull request by gavv**
https://github.com/roc-streaming/roc-toolkit/issues/487

### CODE REVIEW

File: scripts/ci_checks/android/macos.sh [hunk 1]
```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
```
Comment (adrianopol) [inline scripts/ci_checks/android/macos.sh:3]: nit: I prefer to split different types of options (with and without arguments) for more readability: `set -eux -o pipefail`

File: scripts/ci_checks/android/macos.sh [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+action="$1"
+
+build="darwin-x86_64"
+toolchain="$ANDROID_HOME/ndk/${CI_NDK}/toolchains/llvm/prebuilt/$build"
+
+case "$CI_ABI" in
+    "x86_64")
```
Comment (adrianopol) [inline scripts/ci_checks/android/macos.sh:11]: `(x86_64)`
and so on :)

File: scripts/ci_checks/android/macos.sh [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        arch=$CI_ABI
+        host=$CI_ABI-linux-android
```
Comment (adrianopol) [inline scripts/ci_checks/android/macos.sh:13]:
```suggestion
        arch="$CI_ABI"
        host="$CI_ABI-linux-android"
```

and so on

File: scripts/ci_checks/android/macos.sh [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        sysroot="$toolchain/sysroot/usr/lib/arm-linux-androideabi"
+        ;;
+    *) echo "unrecognized android ABI: '$CI_ABI'"; exit 1;;
+esac
+
+export PATH="$ANDROID_HOME/tools/bin:${PATH}"
+export PATH="$ANDROID_HOME/cmdline-tools/latest/bin:${PATH}"
+export PATH="$toolchain/bin:${PATH}"
+
+if [ "$action" == "--install" ]; then
```
Comment (adrianopol) [inline scripts/ci_checks/android/macos.sh:38]: why not to use more secure bash analogues (`[[ ... ]]`) as soon as your are already using bashisms like `==` and arrays?

also, `case` fits better here than `if`, IMHO

File: scripts/ci_checks/android/macos.sh [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    log_lines=800
+
+    # FIXME:
+    #  https://github.com/roc-streaming/roc-toolkit/issues/435
+    #  https://github.com/roc-streaming/roc-toolkit/issues/516
+    tests=( $(find bin/$host$CI_API -name 'roc-test-*' \
+                   -not -name 'roc-test-ctl' -not -name 'roc-test-address') )
+
+    for t in "${tests[@]}"; do
+        filename=$(basename "${t}")
```
Comment (adrianopol) [inline scripts/ci_checks/android/macos.sh:73]:
```suggestion
        filename="$(basename "${t}")"
```

File: scripts/ci_checks/android/macos.sh [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+        logfile="tests/$host$CI_API/$filename.log"
+        adb push $t $target_prefix/$filename 1> /dev/null
```
Comment (adrianopol) [inline scripts/ci_checks/android/macos.sh:76]:
```suggestion
        adb push "$t" "$target_prefix/$filename" 1> /dev/null
```

File: scripts/ci_checks/android/macos.sh [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+fi
+
+if [ "$action" == "--test" ]; then
+    target_prefix="/data/local/tmp"
+
+    adb shell "su 0 mkdir -p ${target_prefix}/lib"
+    adb push "$sysroot/libc++_shared.so" $target_prefix/lib/libc++_shared.so 1> /dev/null
+    adb push "bin/$host$CI_API/libroc.so" $target_prefix/lib/libroc.so 1> /dev/null
+
+    mkdir -p tests/$host$CI_API
```
Comment (adrianopol) [inline scripts/ci_checks/android/macos.sh:63]:
```suggestion
    mkdir -p tests/"$host$CI_API"
```

File: scripts/ci_checks/android/macos.sh [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+          --host=$host$CI_API
+    exit
+fi
+
+if [ "$action" == "--test" ]; then
+    target_prefix="/data/local/tmp"
+
+    adb shell "su 0 mkdir -p ${target_prefix}/lib"
+    adb push "$sysroot/libc++_shared.so" $target_prefix/lib/libc++_shared.so 1> /dev/null
+    adb push "bin/$host$CI_API/libroc.so" $target_prefix/lib/libroc.so 1> /dev/null
```
Comment (adrianopol) [inline scripts/ci_checks/android/macos.sh:61]:
```suggestion
    adb push "$sysroot/libc++_shared.so" "$target_prefix/lib/libc++_shared.so" 1> /dev/null
    adb push "bin/$host$CI_API/libroc.so" "$target_prefix/lib/libroc.so" 1> /dev/null
```

File: scripts/ci_checks/android/macos.sh [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+if [ "$action" == "--build" ]; then
+    scons -Q \
+          --enable-werror \
+          --disable-soversion \
+          --disable-tools \
+          --enable-tests \
+          --build-3rdparty=libuv,openfec,openssl,speexdsp,cpputest \
+          --compiler=clang \
+          --host=$host$CI_API
```
Comment (adrianopol) [inline scripts/ci_checks/android/macos.sh:52]:
```suggestion
          --host="$host$CI_API"
```

File: scripts/ci_checks/android/macos.sh [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+
+        logfile="tests/$host$CI_API/$filename.log"
+        adb push $t $target_prefix/$filename 1> /dev/null
+        adb shell echo
+
+        python3 scripts/scons_helpers/run-with-timeout.py 300 \
+            adb shell "LD_LIBRARY_PATH=${target_prefix}/lib" \
+            "${target_prefix}/${filename} -v" >$logfile 2>/dev/null && \
+            (tail -2 $logfile) || \
+                (error=$?; tail -n $log_lines $logfile; exit $error)
```
Comment (adrianopol) [inline scripts/ci_checks/android/macos.sh:83]:
```suggestion
            adb shell "LD_LIBRARY_PATH=${target_prefix}/lib" \
            "${target_prefix}/${filename} -v" >"$logfile" 2>/dev/null && \
            (tail -2 "$logfile") || \
                (error="$?"; tail -n "$log_lines" "$logfile"; exit "$error")
```

File: scripts/ci_checks/android/macos.sh (hunk 1 above)
Comment (gavv) [reply scripts/ci_checks/android/macos.sh:3]:
```
git grep 'set -euxo pipefail' | wc -l
25
```

but this is the style happened to be used in this project

File: scripts/ci_checks/android/macos.sh (hunk 2 above)
Comment (gavv) [reply scripts/ci_checks/android/macos.sh:11]: as well as this

**Review by adrianopol — APPROVED** [review]

**Pull request merged**